### ISSUE TITLE
c#: only use version info if required

### ISSUE THREAD

**Issue by jsturtevant** [bytecodealliance/wit-bindgen#1078]

**Comment by sano-suguru** [comment]
Proposing to port the `pkg_has_multiple_versions` guard from `crates/c/src/lib.rs::interface_identifier` to `interface_name` in `csharp/src/world_generator.rs` — the sole version-emit site in the C# backend. The fix makes C# match C's behavior exactly (version segment only when multiple versions of the same package coexist).

Before opening a PR: this is a silent default change that breaks source compatibility for users who authored code into versioned namespaces. The C backend precedent suggests "default, no flag" — but wanted to confirm whether a flag is preferred here.

**Comment by yowl** [comment]
I think we still have lots of changes to come and we don't need to worry about breaking code yet.  There is certainly no expectation on my side. So I would go for consistency with c.

**Comment by jsturtevant** [comment]
Agreed breaking changes are fine.  Only using versions when needed is how rust does it too and make it more developer friendly i believe

**Comment by sano-suguru** [comment]
Opened #1635 for this — CI is green, including the C# suites on Linux and Windows.
